operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -45,17 +47,190 @@ LDAP user federation is not currently configured in Keycloak. For example, if it
+        ```
+
+    6. Create `certs.jks`.
+
+        ```bash
+        ncn-m001# podman run --rm -v "$(pwd):/data" dtr.dev.cray.com/library/openjdk:11-jre-slim keytool -importcert \
+        -trustcacerts -file /data/cacert.pem -alias cray-data-center-ca -keystore /data/certs.jks \
+        -storepass password -noprompt
+        ```
+
+    7. Create `certs.jks.b64` by base-64 encoding `certs.jks`.
+
+        ```bash
+        ncn-m001# base64 certs.jks > certs.jks.b64
+        ```
+
+    8.  Inject and encrypt `certs.jks.b64` into `customizations.yaml`.
+
+        ```bash
+        ncn-m001# cat <<EOF | yq w - 'data."certs.jks"' "$(<certs.jks.b64)" | \
+        yq r -j - | /root/site-init/utils/secrets-encrypt.sh | \
+        yq w -f - -i /root/site-init/customizations.yaml 'spec.kubernetes.sealed_secrets.cray-keycloak'
+        {
+            "kind": "Secret",
+            "apiVersion": "v1",
+            "metadata": {
+            "name": "keycloak-certs",
+            "namespace": "services",
+            "creationTimestamp": null
+            },
+            "data": {}
+        }
+        EOF
+        ```
+
+3. Repopulate the keycloak_users_localize and cray-keycloak Sealed Secrets in the customizations.yaml file with the desired configuration.
+
+   1. Check to see if the `generate:` sections of the Sealed Secrets have been populated with encrypted Sealed Secrets already:
+      
+      ```bash
+      ncn-m001# yq read ./customizations.yaml spec.kubernetes.sealed_secrets.keycloak_users_localize
+      apiVersion: bitnami.com/v1alpha1
+      kind: SealedSecret
+      metadata:
+        annotations:
+          sealedsecrets.bitnami.com/cluster-wide: "true"
+        creationTimestamp: null
+        name: keycloak-users-localize
+      spec:
+        encryptedData:
+          ldap_connection_url: AgAmdO19GaLs3Yr9apnJ/JDuQS+6yMC+LlZrPO8g+9UvF2+0X1TifH/bPb0Dw4VMN/2MURx/vvwJE2DTz9yajuW3YJEdwD4o6z/OZ/qLDxu2u+HZSwRnWLWK6ROTBGMP7r0zOdQIDoeeAZw03+d4/UmiBlTlJhl+DJzcS8VbfJV+neNcZ0p5zcM7skqI5NL4teNItHoeuITC2QQ+TRQc/XOrkj3JxvrzFEtEstJz8fXOUXBOwakhRRzUZl9aAYcT6raK3mPQDg14AkM4JVCeku+h6O4OOoOIygC8FzfrXy+LWE93UZjw/0ZM+c6bqOLd7odGto5EylLaV7HS9V8trUPfKExBbYqoRzm+IU9eG3k8Gr7ijT9hRhr1wiV73DFy5NnNB0uAjFClnPXbqntbnSScLxHgFqrnitKM19RzcuHFaZ0Hq0S0VPO8az8BL7jiCkhlxqT0WN6I5RerHPt0PocikKJ/S58a2dc8uGwMgyAPCcJwNXnh7qoA3pgL72kD292ReCYUz6XR51XQAaW1S5O/OaI3VKCCHZAw+qCgW3tCraL3mjzuTYSQQQsDFicxcgQZVz26S/9ATW+3g8btOZlsJ8aA9zpSGAf+uo2N/OKDCmU60fxFTvddcYDlWZ9ZXM1q9lnDTzz1T+QMCZ/f+c2El4CgNvrKJvLueduuagGNfrYoVgGoQ76e/WVi866d1a0=
+        template:
+          metadata:
+            annotations:
+              sealedsecrets.bitnami.com/cluster-wide: "true"
+            creationTimestamp: null
+            name: keycloak-users-localize
+      generate:
+        data: {}
+      ```
+
+      If these sections were populated during the install, proceed to the next sub-step to 
+      remove the existing keycloak_users_localize and cray-keycloak Sealed Secrets
```

Comment:
If step 2 is optional, then I think we need to include a statement to not remove the cray-keycloak secret if it was updated in step 2. If it has not been updated since the install, then it should be removed.

Reply:
How about move step "2. (Optional) Add the LDAP CA certificate in the certs.jks section of customizations.yaml."
after "remove the existing keycloak_users_localize and cray-keycloak Sealed Secrets"?

"2. Update the LDAP settings with the desired configuration." has the admin changing the cray-keycloak/keycloak-certs secret to the empty jks file. If they do that and they need to import the LDAP server CA cert then they'll have to do "2. (Optional) Add the LDAP CA certificate in the certs.jks section of customizations.yaml."